packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+| 300  | Not your turn     | Not your turn to update channel, cannot close |
+| 301  | Channel not found | Could not find channel to update in storage   |
+
+## ChallengeChannel
+
+TODO: (MED) Is there a mechanism to "cancel" challenges?
+Initiates an onchain challenge for a given channel. Will take the currently stored channel state, and put it onchain returning a `ChannelResult` object, and limiting the usage of the channel.
+
+### Parameters
+
+| Name      | Type    | Description               |
+| --------- | ------- | ------------------------- |
+| channelId | Bytes32 | Unique channel identifier |
+
+### Response
+
+| Name                    | Type              | Description                                |
+| ----------------------- | ----------------- | ------------------------------------------ |
+| participants            | `Participant[]`   | Identifying information members of channel |
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+### Errors
+
+| Code | Message           | Description                                    |
+| ---- | ----------------- | ---------------------------------------------- |
+| 1300 | Channel not found | Could not find channel to challenge in storage |
+
+## GetBudget
+
+TODO: (MED) Is this correct?
+Returns the amount available to allocate to different applications within your ledger channel.
+
+### Parameters
+
+| Name             | Type   | Description                               |
+| ---------------- | ------ | ----------------------------------------- |
+| hubParticipantId | string | Participant identifier for ledger channel |
+
+### Response
+
+If the budget cannot be found, this method will return an empty object. Otherwise, it returns a `DomainBudget`:
+
+| Name       | Type          | Description                                                            |
+| ---------- | ------------- | ---------------------------------------------------------------------- |
+| domain     | string        | The channel the funds are attributed to                                |
+| hubAddress | string        | Signing address of channel intermediary                                |
+| budgets    | TokenBudget[] | An array of approved send and received capacity for a list of channels |
+
+### Errors
+
+TODO: (HIGH) Define errors
+
+## GetChannels
```

Comment:
Might make sense to include some query language

Reply:
What type of query language? Like just example code usage?

Reply:
Not sure, but it seems inefficient to have a general purpose API to "get everything" instead of something like... "get everything that is running" or "get everything with this appDefinition". A `Store` might implement this differently depending on the size and situation of the database.

Reply:
That is true, but there are a lot of different `getters` we could settle on. We can look at some of the ones that we have defined in our node, but it's hard to envision a complete set of performant getters needed by every application.

We should definitely think more about the ones that are needed though (i.e. get all by definition, or get all by participant)

Reply:
What about what I just added to the doc in the parameters?